### PR TITLE
Fix watch history order

### LINKED ISSUE #3649: [Enhancement] bump video watched, already in watch history
**Is your enhancement request related to a problem? Please describe.**

When a video is watched and it's already in watch history, it should be bumped to the beginning of the list.
If the video was watched long time ago (several pages into the watch history) it is perceived as if it was not added.
That's why it should be put into the beginning.

**Describe the solution you'd like**

As described 

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by iBicha**
Closes #3649

### CODE REVIEW

**Comment by iBicha** [comment]
TODO: Also, importing data (importing watch history) should take precedence in the order (aka, since you just imported a list of watched videos, then they should be at the top)

File: src/invidious/database/users.cr [hunk 1]
```diff
@@ -50,6 +50,7 @@ module Invidious::Database::Users
   end
 
   def mark_watched(user : User, vid : String)
+    mark_unwatched(user, vid)
```
Comment (SamantazFox) [inline src/invidious/database/users.cr:53]: Can you change the SQL query below to handle everything in a single DB request?

**Review by SamantazFox — CHANGES_REQUESTED** [review]

File: src/invidious/database/users.cr (hunk 1 above)
Comment (iBicha) [reply src/invidious/database/users.cr:53]: I'm not sure how to do that, not the Postgres expert

Comment (SamantazFox) [reply src/invidious/database/users.cr:53]:
```sql
UPDATE users
SET watched = array_append(array_remove(watched, $1), $1)
WHERE email = $2
```

**Comment by iBicha** [comment]
Bump - let me know if there's something else I can help with!

**Comment by SamantazFox** [comment]
PR deployed on test.invidious.io
Sorry for the delay !

**Review by SamantazFox — APPROVED** [review]

**Comment by SamantazFox** [comment]
Thanks for your contribution :D

**Pull request merged**